Sim/SimPapasDetector/src/CMSDetectorSvc.cpp

```diff
@@ -0,0 +1,40 @@
+#include "CMSDetectorSvc.h"
+#include "CMSEcalSvc.h"
+#include "CMSFieldSvc.h"
+#include "CMSHcalSvc.h"
+#include "CMSTrackerSvc.h"
+
+#include "papas/detectors/cms/CMS.h"
+
+DECLARE_SERVICE_FACTORY(CMSDetSvc)
+
+CMSDetSvc::CMSDetSvc(const std::string& name, ISvcLocator* svc) : base_class(name, svc), m_detector(nullptr) {}
+
+StatusCode CMSDetSvc::initialize() {
+  if (Service::initialize().isFailure()) {
+    error() << "Unable to initialize Service()" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  SmartIF<IPapasFieldSvc> papasFieldSvc;
```

Comment:
What is SmartIF?

Reply:
The documentation says:
Small smart pointer class with automatic reference counting for IInterface.

SmartIF simplifies the interaction with components in Gaudi by implementing an automatic reference counting and the casting (via IInterface::queryInterface).

Any the wiser? This is the way other services have been accessed in FCCSW and I copied their approach.

Reply:
ok, thank you.